### PR TITLE
terminal_open: do not force autocmds

### PR DESCRIPTION
**Pull request by justinmk**

### CODE REVIEW

File: src/nvim/terminal.c [hunk 1]
```diff
@@ -236,7 +236,7 @@ Terminal *terminal_open(TerminalOptions opts)
   set_option_value((uint8_t *)"relativenumber", false, NULL, OPT_LOCAL);
   RESET_BINDING(curwin);
   // Apply TermOpen autocmds so the user can configure the terminal
-  apply_autocmds(EVENT_TERMOPEN, NULL, NULL, true, curbuf);
+  apply_autocmds(EVENT_TERMOPEN, NULL, NULL, false, curbuf);
```
Comment (justinmk) [inline src/nvim/terminal.c]: If `true` then this autocmd has "nested" behavior, which is unusual. We also send `false` for TermClosed. cc @tarruda @mhinz

Comment (mhinz) [inline src/nvim/terminal.c]: :+1:

Comment (tarruda) [inline src/nvim/terminal.c]: :+1:

**Comment by watiko** [comment]
After this, `:e term://cmd` does not triggers `TermOpen` because `BufReadCmd` is used for the implement (https://github.com/neovim/neovim/blob/v0.1.2/src/nvim/main.c#L319-L327).

Is it intended behaviour?

**Comment by justinmk** [comment]
main.c needs to be fixed, we shouldn't use BufReadCmd to implement internal functionality.

**Pull request merged**